src/lib.rs

```diff
@@ -256,58 +292,128 @@ fn interface_mtu_windows(socket: &UdpSocket) -> Result<usize, Error> {
 
 #[cfg(test)]
 mod test {
-    use std::net::ToSocketAddrs;
+    use std::{
+        net::{Ipv4Addr, Ipv6Addr, SocketAddr, SocketAddrV4, SocketAddrV6, ToSocketAddrs},
+        sync::atomic::AtomicU16,
+    };
+
+    use crate::interface_mtu;
+
+    cfg_if::cfg_if! {
+        if #[cfg(target_os = "macos")] {const LOCAL_MTU: usize = 16_384;}
+        else if #[cfg(target_os = "linux")] {const LOCAL_MTU: usize = 65_536;}
+        else if #[cfg(target_os = "windows")] { const LOCAL_MTU: usize = 4_294_967_295;}
+        else { panic!("Unsupported platform"); }
```

Comment:
Question: can you make a `const fn platform_mtu() -> usize` that uses `cfg!()` instead?  I am fairly confident that a const function works for that and you can avoid a dependency on `cfg_if`.